ch32v003fun/ch32v003fun.c

```diff
@@ -1321,6 +1321,18 @@ void handle_reset( void )
 
 #endif
 
+#if defined( __riscv_d )
+#define FLOAD( src, offset, dst ) \
+"	fld " #src ", " #offset "*8(" #dst ")\n"
+#define FSTORE( dst, offset, src ) \
+"	fsd " #dst ", " #offset "*8(" #src ")\n"
+#else
+#define FLOAD( src, offset, dst ) \
+"	flw " #src ", " #offset "*4(" #dst ")\n"
+#define FSTORE( dst, offset, src ) \
+"	fsw " #dst ", " #offset "*4(" #src ")\n"
+#endif
+
```

Comment:
if you enable Double support, the Float support is also enabled (aka, F is always defined if D is defined), so bellow I only need to check if F is defined. I wrote it like this to avoid having to put in every single `elif` in the main sections. If someone needs to add Quad Float support, they only need to change it here, the code bellow remains valid and correct.

Reply:
> if you enable Double support, the Float support is also enabled (aka, F is always defined if D is defined),

I didn't think that this was necessarily the case - I thought that F and D could be implemented separately or together?

Reply:
> I didn't think that this was necessarily the case - I thought that F and D could be implemented separately or together?

Sorry, I was wrong: 

- https://github.com/riscv/riscv-isa-manual/blob/main/src/naming.adoc

> Some ISA extensions depend on the presence of other extensions, e.g., "D" depends on "F" and "F" depends on "Zicsr".